handler/src/main/java/io/netty/handler/ssl/SslContextBuilder.java

```diff
@@ -137,14 +187,39 @@ public SslContextBuilder keyManager(File keyCertChainFile, File keyFile) {
      *     password-protected
      */
     public SslContextBuilder keyManager(File keyCertChainFile, File keyFile, String keyPassword) {
+        X509Certificate[] keyCertChain;
+        PrivateKey key;
+        try {
+            keyCertChain = SslContext.toX509Certificates(keyCertChainFile);
+        } catch (Exception e) {
+            throw new IllegalArgumentException("File does not contain valid certificates: " + keyCertChainFile, e);
+        }
+        try {
+            key = SslContext.toPrivateKey(keyFile, keyPassword);
+        } catch (Exception e) {
+            throw new IllegalArgumentException("File does not contain valid private key: " + keyFile, e);
+        }
+        return keyManager(key, keyPassword, keyCertChain);
+    }
+
+    /**
+     * Identifying certificate for this host. {@code keyCertChain} and {@code key} may
+     * be {@code null} for client contexts, which disables mutual authentication.
+     *
+     * @param key a PKCS#8 private key file
+     * @param keyPassword the password of the {@code key}, or {@code null} if it's not
+     *     password-protected
+     * @param keyCertChain an X.509 certificate chain
+     */
+    public SslContextBuilder keyManager(PrivateKey key, String keyPassword, X509Certificate... keyCertChain) {
         if (forServer) {
-            checkNotNull(keyCertChainFile, "keyCertChainFile required for servers");
-            checkNotNull(keyFile, "keyFile required for servers");
+            checkNotNull(keyCertChain, "keyCertChainFile required for servers");
+            checkNotNull(key, "keyFile required for servers");
         }
-        this.keyCertChainFile = keyCertChainFile;
-        this.keyFile = keyFile;
+        this.keyCertChain = keyCertChain.clone();
```

Comment:
You still need to check if `keyCertChain` is `null` or if `keyCertChain` is empty or if `keyCertChain` contains `null`.